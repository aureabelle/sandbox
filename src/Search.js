import React, { Component } from 'react';
import { Table, Select, Alert, Icon } from 'antd';
import fetch from 'node-fetch';

const Option = Select.Option;
const columns = [
  {
    title: 'First Name',
    dataIndex: 'first_name',
  },
  {
    title: 'Last Name',
    dataIndex: 'last_name',
  },
  {
    title: 'Favorite Snack',
    dataIndex: 'fave_snack',
  },
  {
    title: 'Gender',
    dataIndex: 'gender',
  },
  {
    title: 'Email',
    dataIndex: 'email',
  },
  {
    title: 'IP',
    dataIndex: 'ip_address',
  },
  {
    title: 'ID',
    dataIndex: 'id',
  },
];

class Search extends Component {
  constructor(props) {
    super(props);
    this.state = {
      snackers: [],
      products: [],
      selectedSnack: '',
      isSearching: false,
    };
    this.handleSelectChange = this.handleSelectChange.bind(this);
  }

  componentDidMount() {
    const proxyUrl = 'https://cors-anywhere.herokuapp.com/';
    const snackersData = 'https://s3.amazonaws.com/misc-file-snack/MOCK_SNACKER_DATA.json';
    const productsData = 'https://ca.desknibbles.com/products.json?limit=250';

    fetch(proxyUrl + snackersData)
      .then(res => res.json())
      .then(result => {
          this.setState({
            snackers: result,
            isSearching: true
          });
        }, error => {
          this.setState({
            error
          });
        });

    fetch(proxyUrl + productsData)
      .then(response => response.json())
      .then(result => {
        this.setState({
          products: result.products
        });

      }, error => {
        this.setState({
          dataError: error
        });
      });
  }

  handleSelectChange(value) {
    const { selectedSnack, snackers } = this.state;
    this.setState({
      selectedSnack: value
    });
  };

  render() {
    const { snackers, products, selectedSnack } = this.state;
    const filteredSnackers = snackers.filter(snacker => snacker.fave_snack === selectedSnack);
    const snackersList = filteredSnackers.length !== 0 ? filteredSnackers : snackers;
    const existInInventory = snackers.some(snacker => snacker.fave_snack === selectedSnack);

    return (
      <div className="container">
        <h1>Find snackers who like</h1>

        <div className="dropdown">
          <Select
            defaultValue={selectedSnack}
            onChange={this.handleSelectChange}
          >
            <Option value=''>Select Snack</Option>
            {products.map((product, index) => {
              return (
                <Option value={product.title}>{product.title} - {product.variants[0].price}</Option>
              );
            })}
          </Select>
        </div>

        {(!existInInventory && selectedSnack !== '')  &&
          <div className="message">
            <Alert
              message={`Nobody likes ${selectedSnack}.`}
              type="error"
              showIcon
              icon={<Icon type="frown" />}
            />
          </div>
        }

        <Table
          dataSource={snackersList}
          columns={columns}
        />
      </div>
    );
  }
}

export default Search;
