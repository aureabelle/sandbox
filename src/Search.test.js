import React from 'react';
import Enzyme, { shallow, mount, render } from 'enzyme';
import Search from './Search';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

const defaultProps = {
    snackers: [
        {
            email: "gmcwhinnie0@sfgate.com",
            fave_snack: "Hagenes and Sons",
            first_name: "Gerri",
            gender: "Male",
            id: 1,
            ip_address: "208.75.197.92",
            last_name: "McWhinnie"
        },
        {
            email: "dellerey1@ezinearticles.com",
            fave_snack: "Pollich, Mueller and Ratke",
            first_name: "Decca",
            gender: "Male",
            id: 2,
            ip_address: "70.28.165.135",
            last_name: "Ellerey"
        },
        {
            email: "ccleugh2@ezinearticles.com",
            fave_snack: "Kautzer-Feil",
            first_name: "Cicily",
            gender: "Female",
            id: 3,
            ip_address: "74.177.25.215",
            last_name: "Cleugh"
        }
    ]

};

describe('Search Component', () => {
    test('matches the snapshot', () => {
        const wrapper = shallow(<Search />);
        expect(wrapper).toMatchSnapshot();
    });

    test('renders correctly', () => {
        const wrapper = shallow(<Search />);
        expect(wrapper.exists()).toBe(true);
    });

    test('renders the header', () => {
        const wrapper = shallow(<Search />);
        const header = wrapper.find('h1');
        expect(header.text()).toBe('Find snackers who like');
    });

    test('renders the table of snackers', () => {
        const wrapper = render(<Search snackers={defaultProps.snackers} />);
        const table = wrapper.find('Table')[0].children.length;
        expect(table).toBe(3);
    });
});