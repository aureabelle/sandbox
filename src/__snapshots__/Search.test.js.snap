// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component matches the snapshot 1`] = `ShallowWrapper {}`;
